metalearn/metafeatures/base.py

```diff
@@ -0,0 +1,176 @@
+import inspect
+from abc import ABC, abstractmethod
+from typing import List, Callable, Dict, Union, Optional, Any
+import itertools
+
+from metalearn.metafeatures.constants import ProblemType, MetafeatureGroup
+
+class ResourceComputer:
+
+    def __init__(
+        self,
+        computer: Callable,
+        returns: List[str],
+        argmap: Optional[Dict[str,Any]] = {}
+    ) -> None:
+        """
+        Decorates ``computer``, a resource computing function
+        with metadata about that function.
+        
+        Parameters
+        ----------
+        computer
+            The function that computes the resources.
+        returns
+            The names of the resources that ``computer`` returns, specified in
+            the same order as ``computer`` returns them.
+        argmap
+            A custom map of ``computer``'s argument names to the global resource names
+            that will be passed as ``computer``'s arguments when ``computer`` is called.
+        """
+        self._computer = computer
+        self.returns = returns
+
+        self.argmap = {}
+
+        # reversing is needed because `self.defaults` gives the default
+        # argument values corresponding to the *last* `n` arguments in the
+        # function signature.
+        reversed_args = self.args[::-1]
```

Comment:
What is `self.args`? I don't see it defined anywhere, yet you are accessing it here. I do see that everything runs. What is this magic?